### PR TITLE
fix missing statement_info record, after split syntax error statement.

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/8542
issue https://github.com/matrixorigin/matrixone/issues/8168
issue https://github.com/matrixone-cloud/mocloud-services/issues/515#issuecomment-1479470393 - find `use <db>` missing txnID

## What this PR does / why we need it:

bugfix:
1. fix log panic while mo startup
2. fix statement_info's host = src_host
3. fix statement_info's txnID with sql cmd `use <db>`
4. fix missing statement_info record, after split syntax error statement.

### CODE REVIEW

File: pkg/frontend/mysql_cmd_executor.go [hunk 1]
```diff
@@ -241,7 +240,7 @@ var RecordStatement = func(ctx context.Context, ses *Session, proc *process.Proc
 		Account:              tenant.GetTenant(),
 		RoleId:               proc.SessionInfo.RoleId,
 		User:                 tenant.GetUser(),
-		Host:                 sessInfo.GetHost(),
+		Host:                 ses.protocol.Peer(),
```
Comment (xzxiong) [inline pkg/frontend/mysql_cmd_executor.go:243]: fix statement_info's host = src_host

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -405,6 +404,7 @@ func getDataFromPipeline(obj interface{}, bat *batch.Batch) error {
 }
 
 func doUse(ctx context.Context, ses *Session, db string) error {
+	defer RecordStatementTxnID(ctx, ses)
```
Comment (xzxiong) [inline pkg/frontend/mysql_cmd_executor.go:407]: fix statement_info's txnID with sql cmd use <db>

File: pkg/frontend/mysql_cmd_executor.go [hunk 3]
```diff
@@ -265,9 +264,15 @@ var RecordStatement = func(ctx context.Context, ses *Session, proc *process.Proc
...
-	for _, sql := range parsers.SplitSqlBySemicolon(envStmt) {
+var RecordParseErrorStatement = func(ctx context.Context, ses *Session, proc *process.Process, envBegin time.Time, envStmt string, sqlTypes []string, err error) context.Context {
+	retErr := moerr.NewParseError(ctx, err.Error())
+	sqlType := sqlTypes[0]
+	for i, sql := range parsers.SplitSqlBySemicolon(envStmt) {
+		if i < len(sqlTypes) {
+			sqlType = sqlTypes[i]
+		}
 		ctx = RecordStatement(ctx, ses, proc, nil, envBegin, sql, sqlType, true)
+		motrace.EndStatement(ctx, retErr, 0)
```
Comment (xzxiong) [inline pkg/frontend/mysql_cmd_executor.go:275]: normally, it just need to call once in function `logStatementStringStatus`.
but here it generates multi statement_info, so it needs to call `EndStatement` at the same time.

**Review by daviszhen — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Comment by xzxiong** [comment]
@merifyio refresh

**Comment by xzxiong** [comment]
@mergifyio refresh

**Pull request merged**